lib/oidc/mixin/index.ts

```diff
@@ -81,6 +85,20 @@ export function mixinOAuth
 
       // TokenManager
       this.tokenManager = new TokenManager(this, this.options.tokenManager);
+
+      const _setLocation = (url) => {
+        if (this.options.setLocation) {
+          this.options.setLocation(url);
+        } else {
+          window.location = url;
+        }
+      };
+      const enrollAuthenticatorFn = enrollAuthenticator.bind(null, this) as EnrollAuthenticatorFunction;
+      const enrollAuthenticatorApi: EnrollAuthenticatorAPI = Object.assign(enrollAuthenticatorFn, {
+        // This is exposed so we can set window.location in our tests
```

Comment:
this pattern (_setLocation on the function) is very old and I don't think we necessarily need to support it any longer. I believe we have other ways of spying on this action in our tests and we don't want to support this type of interface publicly. 

Since we've never officially supported getWithRedirect._setLocation I think there is an opportunity to refactor both getWithRedirect and the new enrollAuthenticator function to have a cleaner interface, without a `_setLocation` member.